### PR TITLE
Added the Planet Class and simple pressure switch 

### PR DESCRIPTION
**Pull request by CaymanUnterborn**
Still working but able to be critiqued. Updating example as well.

### CODE REVIEW

File: burnman/planet.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+# GPL v2 or later.
+
+import numpy as np
+from scipy.integrate import odeint
+from scipy.integrate import quad
+from scipy.interpolate import UnivariateSpline
+import burnman
+from burnman import minerals as minerals
+
+G = 6.67e-11
```
Comment (tjhei) [inline burnman/planet.py]: remove G

File: burnman/planet.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
+from __future__ import print_function
+# This file is part of BurnMan - a thermoelastic and thermodynamic toolkit for the Earth and Planetary Sciences
+# Copyright (C) 2012 - 2015 by the BurnMan team, released under the GNU
+# GPL v2 or later.
+
+import numpy as np
+from scipy.integrate import odeint
+from scipy.integrate import quad
+from scipy.interpolate import UnivariateSpline
+import burnman
```
Comment (tjhei) [inline burnman/planet.py]: remove this line

File: burnman/planet.py [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+    def __init__(self, radii, temperatures, compositions):
+
+        self.radii = radii
+        self.temperatures = temperatures
+        self.pressures = np.linspace(350.0e9, 0.0, len(radii)) # initial guess at pressure profile
+        self.compositions = compositions
+        #creates the geotherm, starting with just 300 K isothermal
+        self.averaging_scheme = burnman.averaging_schemes.VoigtReussHill()
+
+    def generate_profiles( self, radii, n_iterations ):
```
Comment (tjhei) [inline burnman/planet.py]: don't do a fixed number of iterations. Call argument ``depths`` or something so we don't confuse them.

File: burnman/planet.py [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+        #Generate the density, gravity, pressure, vphi, and vs profiles for the planet.
```
Comment (tjhei) [inline burnman/planet.py]: use ``"""`` to make function comments

File: burnman/planet.py (hunk 3 above)
Comment (tjhei) [inline burnman/planet.py]: Is there a reason ``generate_profiles`` is not automatically called when ``Planet`` is constructed?

File: burnman/planet.py [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+            self.gravity = self._compute_gravity(self.densities, self.radii)
+            self.pressures = self._compute_pressure(self.densities, self.gravity, self.radii)
+
+        self.mass = self._compute_mass(self.densities, self.radii)
+
+        self.moment_of_inertia = self._compute_moment_of_inertia(self.densities, self.radii)
+        self.moment_of_inertia_factor = self.moment_of_inertia / self.mass / self.radii[-1] / self.radii[-1]
+
+    def _evaluate_eos(self, compositions, pressures, temperatures, radii,averaging_scheme):
+        #Evaluates the equation of state for each radius slice of the model.
```
Comment (tjhei) [inline burnman/planet.py]: same here, make ``""" ... """`` style comments

Comment (tjhei) [inline burnman/planet.py]: remove ``averaging_scheme`` argument and use ``self.averaging_scheme``

File: burnman/planet.py [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+                                 radii[0], radii[-1] )[0]
+        return moment
+
+
+def compute_shell_mass( density, radii):
+    #returns a mass for a layer or sum of layers smaller than the whole planet
+        rhofunc = UnivariateSpline(radii, density )
+        mass = quad( lambda r : 4.*np.pi*rhofunc(r)*r*r,
+                                 radii[0], radii[-1] )[0]
+        return mass
```
Comment (tjhei) [inline burnman/planet.py]: missing newline at end of file

File: burnman/planet.py [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+    def _compute_moment_of_inertia( self, density, radii):
+        #Returns the moment of inertia of the planet [kg m^2]
+
+        rhofunc = UnivariateSpline(radii, density )
+        moment = quad( lambda r : 8.0/3.0*np.pi*rhofunc(r)*r*r*r*r,
+                                 radii[0], radii[-1] )[0]
+        return moment
+
+
+def compute_shell_mass( density, radii):
```
Comment (tjhei) [inline burnman/planet.py]: I am confused by this function. How do you supply the shell radius?

File: burnman/planet.py [hunk 8]
```diff
@@ -0,0 +1,216 @@
...
+
+        # initial pressure guess: linear as a function of radius
+        max_p = 350e9
+        self.pressures = np.array([max_p * (1.0 - r / self.radius) for r in self.radial_slices])
+        self.temperatures = [self.get_layer_by_radius(r).temperature(r) for r in self.radial_slices]
+        self.densities = np.empty_like(self.pressures)
+        self.gravity = np.empty_like(self.pressures)
+        self.bulk_sound_speed = np.empty_like(self.pressures)
+        self.shear_velocity = np.empty_like(self.pressures)
+
```
Comment (sannecottaar) [inline burnman/planet.py]: Remove seismic velocities here and in _evaluate_eos. These can be computed after the pressure-density profile is made self-consistent.

File: burnman/planet.py [hunk 9]
```diff
@@ -0,0 +1,216 @@
...
+        raise LookupError()
+
+    def _evaluate_eos(self):
+        # evaluate each layer separately
+        for layer in self.layers:
+            mypressures = self.pressures[layer.n_start: layer.n_end]
+            mytemperatures = self.temperatures[layer.n_start: layer.n_end]
+
+            density, vs, vphi = layer.rock.evaluate(['density', 'v_s', 'v_phi'], mypressures, mytemperatures)
+
```
Comment (sannecottaar) [inline burnman/planet.py]: density = ....evaluate(['density'],

File: burnman/planet.py [hunk 10]
```diff
@@ -0,0 +1,216 @@
...
+            self.densities[layer.n_start: layer.n_end] = density
+            self.bulk_sound_speed[layer.n_start: layer.n_end] = vs
+            self.shear_velocity[layer.n_start: layer.n_end] = vphi
+
```
Comment (sannecottaar) [inline burnman/planet.py]: add a function to evaluate any properties
def evaluate(self, variables, pressures= none, depths=none):

If no pressures or depths are defined it returns along the internal depth list.

**Comment by tjhei** [comment]
merging. We can fix things later on.

**Pull request merged**